modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -2198,8 +2198,24 @@ public void repopulate() throws IndexRebuildException {
               }
               current++;
 
-              var updatedWorkflowData = index.getEvent(indexData.getMediaPackageId(), indexData.getOrganizationId(),
-                        securityService.getUser());
+              String orgid = indexData.getOrganizationId();
+              if (null == orgid) {
+                //We're assuming here that mediapackages don't change orgs
+                orgid = assetManager.getSnapshotsById(indexData.getMediaPackageId())
```

Comment:
We should check if the result is empty here, in case the media package associated with the workflow doesn't exist anymore.

Reply:
Good catch, added.